django/thunderstore/api/cyberstorm/views/user.py

```diff
@@ -0,0 +1,45 @@
+from django.contrib.auth import get_user_model
+from rest_framework import serializers
+from rest_framework.exceptions import PermissionDenied, ValidationError
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from thunderstore.api.utils import conditional_swagger_auto_schema
+from thunderstore.social.views import DeleteAccountForm
+
+User = get_user_model()
+
+
+class CyberstormUserDeleteRequestSerialiazer(serializers.Serializer):
+    verification = serializers.CharField()
+
+
+class CyberstormUserDeleteResponseSerialiazer(serializers.Serializer):
+    username = serializers.CharField()
+
+
+class UserDeleteAPIView(APIView):
+    @conditional_swagger_auto_schema(
+        request_body=CyberstormUserDeleteRequestSerialiazer,
+        responses={200: CyberstormUserDeleteResponseSerialiazer},
+        operation_id="cyberstorm.user.delete",
+        tags=["cyberstorm"],
+    )
+    def post(self, request, username, format=None):
+        serializer = CyberstormUserDeleteRequestSerialiazer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        if request.user.username != username:
+            raise PermissionDenied("You can only delete your own account")
```

Comment:
Given this constraint exists, I feel like this view should not take the URL path parameter at all and instead just treat it as one of the "current user" APIs that are around, e.g. `/current-user/delete/`. No need to add this condition at all